tuner/tuner/common.py

```diff
@@ -105,26 +110,63 @@ def is_comptible(mma_intrinsic: iree_gpu.MMAIntrinsic) -> bool:
     return list(filter(is_comptible, mma_intrinsics))
 
 
-class ReorderWorkgroupsStrategy(Enum):
-    NONE = 0
-    SWIZZLE = 1
-    TRANSPOSE = 2
-
-    def __str__(self) -> str:
-        return self.name.title()
-
-
 @dataclass
 class Configuration:
     subgroup_size: int
     workgroup_size: list[int]
-    intrinsic: iree_gpu.MMAAttr
-    tile_sizes: list[int]
-    subgroup_m_count: int
-    subgroup_n_count: int
+    lowering_config: iree_gpu.LoweringConfigAttr
     gpu_pipeline_options: iree_gpu.PipelineOptionsAttr
     waves_per_eu: int
 
+    def intrinsic(self) -> Optional[iree_gpu.MMAAttr]:
+        if self.lowering_config.attributes.__contains__("mma_kind"):
+            return self.lowering_config.attributes.__getitem__("mma_kind")
+        return None
+
+    def tilesize_workgroup(self) -> list[int]:
+        if self.lowering_config.attributes.__contains__("workgroup"):
+            workgroup_attrs = self.lowering_config.attributes.__getitem__("workgroup")
+            return [attr.value for attr in workgroup_attrs]
+        return []
+
+    def tilesize_reduction(self) -> list[int]:
+        if self.lowering_config.attributes.__contains__("reduction"):
+            reduction_attrs = self.lowering_config.attributes.__getitem__("reduction")
+            return [attr.value for attr in reduction_attrs]
+        return []
+
+    def subgroup_m_count(self) -> Optional[int]:
+        if self.lowering_config.attributes.__contains__("subgroup_m_count"):
+            attr = self.lowering_config.attributes.__getitem__("subgroup_m_count")
+            return attr.value
+        return None
+
+    def subgroup_n_count(self) -> Optional[int]:
+        if self.lowering_config.attributes.__contains__("subgroup_n_count"):
+            attr = self.lowering_config.attributes.__getitem__("subgroup_n_count")
+            return attr.value
+        return None
+
+
+def get_lowering_config(
```

Comment:
Please add tests for this and make sure the returned attributes roundtrip with the helpers from `Configuration`